sklego/preprocessing.py

```diff
@@ -298,30 +298,38 @@ def fit(self, X, y=None):
         :raises:
             ``ValueError`` if ``X`` contains non-numeric columns
         """
+        # Attempts to avoid copying X
         X = check_array(X, copy=False, force_all_finite=False, dtype=FLOAT_DTYPES, estimator=self)
 
-        # Saving the number of columns to ensure coherence between fit and transform inputs
-        self.n_columns_ = X.shape[1]
-
-        # Converting to pandas.DataFrame because:
-        # 1. It knows how to deal with inf filters
-        # 2. It can compute quantiles even in the presence of nan values
-        # numpy arrays can't do any of the above
-        X = pd.DataFrame(X)
-
-        # Making sure that the magnitudes of -np.inf and np.inf won't cause any trouble
-        X = X[(-np.inf < X) & (X < np.inf)]
-
-        # Computing the quantiles for each column of X
-        min_quantile, max_quantile = self.quantile_range
-        quantiles = X.quantile([min_quantile/100, max_quantile/100], interpolation=self.interpolation)
-
-        # There shouldn't be any nan cells in self.quantiles_. If that's not the case, it means that
-        # the user asked ColumnCapper to fit columns containing only nan or inf cells.
-        if quantiles.isna().sum().sum() > 0:
+        infs_mask = (X == np.inf) | (X == -np.inf)
+
+        # If X contains infs, we will need to replace them by nans before computing quantiles
+        if infs_mask.any():
+            if isinstance(X, pd.DataFrame):
+                # Copying X is not enough in this case. We need to go deeper.
+                X = X.values.copy()
+            else:
+                X = X.copy()
+            copied = True
+            # Now we're free to edit X
+            np.putmask(X, infs_mask, np.nan)
+        else:
```

Comment:
this is a bit ... nit picky. but technically you do not need the `else`. line 317 could be moved just before the if statement, no?

Reply:
Yes! Also, I'm curious about how much memory the `overwrite_input` saves. If it's not a lot, I might do away with the flag entirely. If it is a lot it might be good to copy `X` regardless whether there are infs or not

Reply:
I'd guess it can save at maximum as much memory as `X` consumes. For the sake of simplicity, I've made it copy `X` regardless.